contracts/test/unit/VoltV2GovCompatibility.t.sol

```diff
@@ -0,0 +1,196 @@
+
+    VoltV2 private voltV2;
+    ICore private core;
+    MockDAO private mockDAO;
+    MockERC20 private mockToken;
+    TimelockController private timelock;
+
+    VoltTestAddresses public addresses = getAddresses();
+
+    address proposerCancellerExecutor = address(0x123);
+    address userWithVolt = address(0xFFF);
+
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    uint256 public quorum = 1_000_000e18;
+
+    function setUp() public {
+        core = getCore();
+
+        address[] memory proposerCancellerAddresses = new address[](1);
+        proposerCancellerAddresses[0] = proposerCancellerExecutor;
+
+        address[] memory executorAddresses = new address[](1);
+        executorAddresses[0] = proposerCancellerExecutor;
+
+        timelock = new TimelockController(
+            600,
+            proposerCancellerAddresses,
+            executorAddresses
+        );
+
+        voltV2 = new VoltV2(address(core));
+        mockDAO = new MockDAO(IVotes(address(voltV2)), timelock);
+
+        timelock.grantRole(timelock.PROPOSER_ROLE(), address(mockDAO));
+        timelock.grantRole(timelock.EXECUTOR_ROLE(), address(mockDAO));
+
+        mockToken = new MockERC20();
+        mockToken.mint(address(timelock), 1_000_000e18);
+    }
+
+    function testUserWithNoVoltCanPropose() public {
+        (
+            address[] memory targets,
+            uint256[] memory values,
+            bytes[] memory calldatas,
+            string memory description,
+
+        ) = _createDummyProposal();
+
+        uint256 noVoltProposalId = mockDAO.propose(
+            targets,
+            values,
+            calldatas,
+            description
+        );
+
+        assertEq(uint8(mockDAO.state(noVoltProposalId)), 0); // Pending
+        vm.roll(block.number + 1);
+
+        assertEq(uint8(mockDAO.state(noVoltProposalId)), 1); // Active
+    }
+
+    function testProposalExecutes() public {
+        (
+            address[] memory targets,
+            uint256[] memory values,
+            bytes[] memory calldatas,
+            string memory description,
+            bytes32 descriptionHash
+        ) = _createDummyProposal();
+
+        vm.prank(addresses.minterAddress);
+        voltV2.mint(userWithVolt, quorum);
+
+        vm.prank(userWithVolt);
+        voltV2.delegate(userWithVolt);
+
+        uint256 proposalId = mockDAO.propose(
+            targets,
+            values,
+            calldatas,
+            description
+        );
+
+        // Advance past the 1 voting block
+        vm.roll(block.number + 1);
+
+        // Cast a vote for the proposal, in excess of quorum
+        vm.prank(userWithVolt);
+        mockDAO.castVote(proposalId, 1);
+
+        vm.roll(block.number + 2);
+
+        vm.startPrank(proposerCancellerExecutor);
+        mockDAO.queue(targets, values, calldatas, descriptionHash);
+        vm.stopPrank();
+
+        vm.warp(block.number + timelock.getMinDelay());
+
+        vm.startPrank(proposerCancellerExecutor);
```

Comment:
adjust to all be under one `startPrank`

Reply:
add assertions that correct number of votes have been cast, add invariant test with multiple users voting

Reply:
fuzz addresses and amounts constrained to quorum, make all cast votes and make sure it still doesn't pass